backends/candle/src/models/modernbert.rs

```diff
@@ -0,0 +1,795 @@
+                        None,
+                        None,
+                        None,
+                        None,
+                    )?;
+
+                    context_layer.reshape((
+                        batch_size,
+                        self.num_attention_heads,
+                        seq_len,
+                        self.attention_head_size,
+                    ))
+                }
+                #[cfg(not(feature = "cuda"))]
+                {
+                    candle::bail!("`cuda` feature is not enabled")
+                }
+            } else {
+                let attn_weights = query_layer.matmul(&key_layer.t()?)?;
+                let attn_weights = (attn_weights * self.softmax_scale)?;
+                let attn_weights = attn_weights.add(attention_mask)?;
+                let attn_weights = candle_nn::ops::softmax_last_dim(&attn_weights)?;
+                attn_weights.matmul(&value_layer.contiguous()?)
+            }?;
+
+        let hidden_states = context_layer.transpose(1, 2)?.flatten_from(D::Minus2)?;
+
+        self.wo.forward(&hidden_states)
+    }
+}
+
+struct ModernBertEncoderLayer {
+    attn_norm: Option<LayerNorm>,
+    attn: ModernBertAttention,
+    mlp_norm: LayerNorm,
+    mlp: ModernBertMLP,
+
+    span: tracing::Span,
+}
+
+impl ModernBertEncoderLayer {
+    pub fn load(vb: VarBuilder, index: usize, config: &ModernBertConfig) -> Result<Self> {
+        let attn_norm = if index != 0 {
+            Some(LayerNorm::load(
+                vb.pp("attn_norm"),
+                config.hidden_size,
+                config.norm_eps as f32,
+            )?)
+        } else {
+            None
+        };
+
+        let attn = ModernBertAttention::load(vb.pp("attn"), config)?;
+
+        let mlp_norm = LayerNorm::load(
+            vb.pp("mlp_norm"),
+            config.hidden_size,
+            config.norm_eps as f32,
+        )?;
+        let mlp = ModernBertMLP::load(vb.pp("mlp"), config)?;
+
+        let span = tracing::span!(tracing::Level::TRACE, "layer");
+
+        Ok(ModernBertEncoderLayer {
+            attn_norm,
+            attn,
+            mlp_norm,
+            mlp,
+            span,
+        })
+    }
+
+    fn forward(
+        &self,
+        hidden_states: &Tensor,
+        attention_mask: &Tensor,
+        cos: &Tensor,
+        sin: &Tensor,
+    ) -> Result<Tensor> {
+        let _enter = self.span.enter();
+
+        let residual = hidden_states.clone();
+
+        let attn_norm = if let Some(attn_norm) = &self.attn_norm {
+            attn_norm.forward(hidden_states, None)?
+        } else {
+            hidden_states.clone()
+        };
+
+        let attn_outputs = self.attn.forward(&attn_norm, attention_mask, cos, sin)?;
+
+        let hidden_states = residual.add(&attn_outputs)?;
+
+        let mlp_output = self
+            .mlp
+            .forward(&self.mlp_norm.forward(&hidden_states, None)?)?;
+
+        hidden_states.add(&mlp_output)
```

Comment:
Same here as mentioned above:

```suggestion
        let hidden_states = hidden_states.add(&mlp_output)?;
        
        Ok(hidden_states)
```